pkg/controller/core/workload_controller.go

```diff
@@ -380,6 +380,7 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	// at this point we know a Workload has at least one Retry AdmissionCheck
 	message := "At least one admission check is false"
 	workload.SetEvictedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, message)
+	workload.SetAllChecksToPending(wl, "AdmissionCheck pending after retry")
```

Comment:
It is not necessarily "after retry" as ready checks can also be flipped. I would move the message construction under the function as it potentially could differ.

Reply:
Done